plugins/destination/postgresql/client/migrate.go

```diff
@@ -3,210 +3,326 @@ package client
+func (c *Client) normalizeTable(table *schema.Table) *schema.Table {
+	switch c.pgType {
+	case pgTypeCockroachDB:
+		return c.normalizeTableCockroach(table)
+	case pgTypePostgreSQL:
+		return c.normalizeTablePg(table)
+	default:
+		panic("unknown pg type")
 	}
+}
 
-	stalePks := c.getStalePks(pgPKs, table)
-	constraintName := pgx.Identifier{table.Name + "_cqpk"}.Sanitize()
-	dropConstraintSQL := "alter table " + tableName + " drop constraint if exists " + constraintName
-	if len(stalePks) > 0 {
-		message := "the following primary keys were removed from the schema %q for table %q.\nYou can migrate the table manually by running:\n%s"
-		sep := strings.Repeat("-", len(dropConstraintSQL)+1)
-		query := fmt.Sprintf("%s\n%s;\n%s\n%s", sep, dropConstraintSQL, getDropNotNullQuery(table, stalePks), sep)
-		return fmt.Errorf(message, stalePks, table.Name, query)
-	}
-
-	reCreatePrimaryKeys := false
-
-	for _, col := range table.Columns {
-		columnName := pgx.Identifier{col.Name}.Sanitize()
-		columnType := c.SchemaTypeToPg(col.Type)
-		if columnType == "" {
-			c.logger.Warn().Str("table", table.Name).Str("column", col.Name).Msg("Column type not supported, skipping")
+func (c *Client) autoMigrateTable(ctx context.Context, table *schema.Table, changes []schema.TableColumnChange) error {
+	for _, change := range changes {
+		switch change.Type {
+		case schema.TableColumnChangeTypeAdd:
+			if err := c.addColumn(ctx, table.Name, change.Current); err != nil {
+				return err
+			}
+		case schema.TableColumnChangeTypeRemove:
 			continue
+		default:
+			panic("unknown change type")
 		}
-		pgColumn := pgColumns.getPgColumn(col.Name)
-
-		switch {
-		case pgColumn == nil:
-			c.logger.Info().Str("table", table.Name).Str("column", col.Name).Msg("Column doesn't exist, creating")
-
-			sql := "alter table " + tableName + " add column " + columnName + " " + columnType
-			if col.CreationOptions.PrimaryKey {
-				reCreatePrimaryKeys = true
-			}
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to add column %s on table %s: %w", col.Name, table.Name, err)
-			}
-		case pgColumn.typ != columnType:
-			c.logger.Info().Str("table", table.Name).Str("column", col.Name).Str("old_type", pgColumn.typ).Str("new_type", columnType).Msg("Column exists but type is different, re-creating")
-			// column exists but type is different
+	}
+	return nil
+}
 
-			// if this column contains primary key we will need to recreate the primary key
-			if c.enabledPks() && col.CreationOptions.PrimaryKey {
-				reCreatePrimaryKeys = true
-			}
-			sql := "alter table " + tableName + " drop column " + columnName
-			// right now we will drop the column and re-create. in the future we will have an option to automigrate
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to drop column %s on table %s: %w", col.Name, table.Name, err)
+func (*Client) canAutoMigrate(changes []schema.TableColumnChange) bool {
```

Comment:
```suggestion
func canAutoMigrate(changes []schema.TableColumnChange) bool {
```

I'm not sure this needs to be a `Client` method

Reply:
I think it can in case we will want to add some logs.